### DISCUSSION TITLE
use case for WebSockets + REST

### DISCUSSION THREAD

**Discussion by huzaifa-99** [hagopj13/node-express-boilerplate#84] [General]
Hi @hagopj13. I hope you are doing well. First of all, I really appreciate the boilerplate and totally love how the structure is defined. I also like how the business logic is separated in service layer. 

For me, I have a use case where I want to emit events from the server to the client, ideally through WebSockets(or SSE) and I also want to use REST endpoints.

- There would be an endpoint that just returns relevant data to that single client instance that requests for it. (Client Request data from a Tab and Server sends data to that specific client tab from the same domain).

- And then there would be an endpoint that pushes/emits data to all open WebSockets or client instances. (Client Request data from a Tab and Server sends data to all client tabs that point to the same domain).

How should the Websockets implementation be structured. I would like to know your thoughts on this use case.

Once again, I appreciate your work.
Thanks.

**Comment by hagopj13** [comment]
Hi @huzaifa-99, I'm glad you like the boilplate and find it useful.

Your use case sounds very similar to #15, I suggest you look there. There are some good ideas there that you can benefit from. Let me know if face any issues or need some help.
